pkg/client/client.go

```diff
@@ -83,6 +106,10 @@ func (c *Client) setupCtx(ctx context.Context, minTimeout time.Duration) (contex
 	return context.WithTimeout(ctx, timeout)
 }
 
+func (c *Client) setConfig(config rest.Config) {
+	c.Config = config
```

Comment:
Since we removed the nil check and now pass by value, we don't need this function we can just set c.Config directly in `ForResourceKind(...)`